make UI of render intro page

shubhamsigdar1 opened the pull request:
### Issue
Open #431 

### What is the change?
- Make UI of intro page 
- created 4 file (intro.html, intro.js, intro.css, utils.js,  img/intro page)
- I get userId from browser and insert into API to fetch details of join form from backend
- fetch `https://api.realdevsquad.com/users/:userId/intro` Data into UI
- only superuser can access this UI
- if superuser go to this link `http://www.realdevsquad.com/intro?id=rvQTbRBHiXh8YrKiKGj4`  will get intro UI 
- add 404 page if superuser write wrong url and userid
- add 401 page when userId access by normal user 
- added toast error message when backend is not running and api is not fetch also when superuser write wrong userId with different message
- In utils.js file i added global function
- converted all images in my pr into svg format
   1. `img/intro-page/exclamation-circle.svg, page-not-authorized.svg, page-not-found.svg`
   2. `img/github.svg , Real-Dev-Squad@1x.svg`

### Is Development Tested?
-No

###  After Change Screenshots/video

### intro-page UI
https://user-images.githubusercontent.com/107163260/228812084-79506418-0856-4d70-beb6-1119046a795f.mp4

### Add 404 page(url access by super user)
https://user-images.githubusercontent.com/107163260/229954855-50811126-1537-4328-a6e5-d90f9ebf4a4c.mp4

### Add 401 page(url access by normal user)
https://user-images.githubusercontent.com/107163260/229954507-57560fd1-f479-4bc8-a443-79367b5ca3da.mp4

### added toast when backend is not running and API is not fetch
https://user-images.githubusercontent.com/107163260/228958347-5c479308-49c4-4657-905f-1850a7247ec8.mp4

RitikJaiswal75 commented:
`/intro.html` wasn't the expected page what if I want to view the intro of some other user how will I do that?
the route was expected to be `/intro/<userId>`

The API route is correct apart from this: `rvQTbRBHiXh8YrKiKGj4` this can be changed to any user's id.

sahsisunny commented on intro.html:
```diff
@@ -0,0 +1,81 @@
+      href="https://fonts.googleapis.com/css2?family=Roboto:wght@100;300;400;500;700;900&display=swap"
+      rel="stylesheet"
+    />
+
+    <!-- Primary Meta Tags -->
+    <meta name="title" content="Real Dev Squad | Intro" />
+    <meta
+      name="description"
+      content="Page for verification of user to join Real Dev Squad."
+    />
+
+    <!-- Open Graph / Facebook Meta Tags -->
```
It is required ? because it is already in index.html.

sahsisunny commented on intro.html:
```diff
@@ -0,0 +1,81 @@
+  </head>
+
+  <body>
+    <div class="not-access"></div>
```
can you show better and different messages for different status codes(401 and 404) ?

shmbajaj commented on css/intro.css:
```diff
@@ -0,0 +1,82 @@
+  }
+}
+
+@media (min-width: 500px) and (max-width: 1024px) {
```
We don't need two media queries here.
This media query styles has to be default styles.

shubhamsigdar1 replied:
not understand @shmbajaj what u said above

shmbajaj replied:
Yes right @sahsisunny, we don't need two media queries.
Either mobile or table/desktop styles have to be default one.

shmbajaj commented on css/intro.css:
```diff
@@ -0,0 +1,82 @@
+    font-size: 1.25rem;
+  }
+
+  .not-found-img {
```
`Alternate`

```css
.not-found-img,
.not-authorized-img{
width: 40%;
}

shmbajaj commented on css/intro.css:
```diff
@@ -0,0 +1,82 @@
+    line-height: 32px;
+  }
+
+  .not-authorized-text-h1 {
```
These styles can also be combined.

shmbajaj replied:
Try not to use tag names in css classes names.
`.not-authorized-text` is enough.

shmbajaj commented on css/intro.css:
```diff
@@ -0,0 +1,82 @@
+  text-align: center;
+}
+
+.not-found {
```
Repeating styles can be combined together.
Here `.not-found` and `.not-authorized`

```css
.not-found,
.not-authorized
{
//css code
}
```

shmbajaj commented on js/intro.js:
```diff
@@ -0,0 +1,320 @@
+      credentials,
+      ...options,
+    });
+    return response;
```
Convert the response to json format from here and then return.

```js
await response.toJson()
```

shubhamsigdar1 replied:
> Convert the response to json format from here and then return.
> 
> ```js
> await response.toJson()
> ```

not able to do because i am using response.status in code and makeApiCall() fetch my request

https://github.com/Real-Dev-Squad/website-www/pull/432/files/f6c2354ca25c52308fa7baf469bbba3e4bb46107#diff-b8034978785404c0d5db1ed2aa00f41fecf7baaf67f71b348a0a1806a5b24017R258

sahsisunny replied:
Use 
```js
await response.json();
```
instead of below `toJson()` method does not exist in Javascript
```js
await response.toJson()
```

shmbajaj replied:
> > Convert the response to json format from here and then return.
> > ```js
> > await response.toJson()
> > ```
> 
> not able to do because i am using response.status in code and makeApiCall() fetch my request
> 
> https://github.com/Real-Dev-Squad/website-www/pull/432/files/f6c2354ca25c52308fa7baf469bbba3e4bb46107#diff-b8034978785404c0d5db1ed2aa00f41fecf7baaf67f71b348a0a1806a5b24017R258

You can send response as object/record having following properties:

```jsx
{
data: any,
error: message<string>,
code: number
}
```

shmbajaj commented on js/intro.js:
```diff
@@ -0,0 +1,320 @@
+}
+
+function getUserId() {
+  const currentUrl = window.location.href;
```
```js
const urlParams = new URLSearchParams(window.location.search);
urlParams.get('userId');
```

You have to do something similar to this, please check **URLSearchParams** over the mdn.

shmbajaj replied:
This is a possible way, but might not be a good way.

shubhamsigdar1 replied:
I will do this after talk to ankush that he need in queryparameter userId or not...right now he is guiding me how to make route in morning session

shmbajaj commented on js/intro.js:
```diff
@@ -0,0 +1,320 @@
+//making userSavedData object from API
+async function showSavedDetails() {
+  try {
+    const userId = getUserId();
```
Do check the above comment related to **URLSearchParams** and make changes here.

shmbajaj commented on js/intro.js:
```diff
@@ -0,0 +1,320 @@
+      }
+    }
+  } catch (err) {
+    alert('something went wrong');
```
Can we use Toast here?

sahsisunny commented on css/intro.css:
```diff
@@ -0,0 +1,82 @@
+  height: 10.25rem;
+}
+
+#renderHeardAbout {
```
Where is the changes @shubhamsigdar1 ?(Please not resolve the conversation if changes not done or without your reply /comment)

sahsisunny commented on js/constants.js:
```diff
@@ -261,6 +261,5 @@ export const JOIN_POST_URL = `${BASE_URL}/users/self/intro`;
   'https://github.com/login/oauth/authorize?client_id=23c78f66ab7964e5ef97';
-export const SIGNUP_URL =
-  'https://my.realdevsquad.com/new-signup';
+export const SIGNUP_URL = 'https://my.realdevsquad.com/new-signup';
```
can you revert it ?

shubhamsigdar1 replied:
it happen during fixing lint not able to revert it

sahsisunny replied:
https://dev.to/lofiandcode/git-and-github-how-to-revert-a-single-file-dha

shubhamsigdar1 replied:
when i am running npm run fix it will change to this...not resolved

FMK2312 replied:
wait, but what is wrong? other than the formatting?

shubhamsigdar1 replied:
nothing

FMK2312 replied:
exactly

sahsisunny commented on intro.html:
```diff
@@ -0,0 +1,133 @@
+      href="https://fonts.googleapis.com/css2?family=Roboto:wght@100;300;400;500;700;900&display=swap"
+      rel="stylesheet"
+    />
+    <script
+      src="https://kit.fontawesome.com/34f93e6da6.js"
+      crossorigin="anonymous"
+    ></script>
```
I think you should first ask whether you can use **fontawesome** icon library or not.

shubhamsigdar1 replied:
asked to rohan-raj he said we can use

sahsisunny commented on js/intro.js:
```diff
@@ -0,0 +1,335 @@
+      ),
+    );
+    setTimeout(function () {
+      window.location.href = 'https://realdevsquad.com';
```
can you use `HOME_URL` constants here ?

sahsisunny commented on css/intro.css:
Can you use REM for size?

shubhamsigdar1 replied:
for height and width??

sahsisunny replied:
> for height and width??

whenever we used PX

sahsisunny commented on js/intro.js:
```diff
@@ -0,0 +1,335 @@
+const queryString = window.location.search;
+const urlParams = new URLSearchParams(queryString);
+
+async function makeApiCall(
+  url,
+  method = 'get',
+  body = null,
+  credentials = 'include',
+  headers = { 'content-type': 'application/json' },
+  options = null,
+) {
+  try {
+    const response = await fetch(url, {
+      method,
+      body,
+      headers,
+      credentials,
+      ...options,
+    });
+    const res = {
+      data: await response.json(),
+      status: response.status,
+    };
+    return res;
+  } catch (err) {
+    throw err;
+  }
+}
+
```
Can you move this function in utility file? so we can use this function in every place .

sahsisunny commented on js/intro.js:
When error is 404 and 401 which is show message or toast ?

shubhamsigdar1 replied:
Is I am right?? you talking about last video of description(i.e. added toast when api is not fetch)->Should i have to add 404 page??
or talking about 2nd video(i.e. Add 404 page(url access by super user)) when superuser write wrong userId its show message??

Please correct me??

sahsisunny replied:
I am not talking about any videos. i just want to know when error messages or toast show ?

shubhamsigdar1 replied:
when backend is not running and api is not fetched

sahsisunny commented:
Great work @shmbajaj and thank you for your efforts in addressing the feedback we provided. I am glad to see that you have made the necessary changes to improve the quality of the code. I am happy to approve your PR. Before approving, I would like to ask you to please reply on the last some of the comments I made. I would like to know your thoughts on those comments. Once you have replied to those comments, I will approve your PR.🎉🎉

shubhamsigdar1 commented:
> Great work @shmbajaj and thank you for your efforts in addressing the feedback we provided. I am glad to see that you have made the necessary changes to improve the quality of the code. I am happy to approve your PR. Before approving, I would like to ask you to please reply on the last some of the comments I made. I would like to know your thoughts on those comments. Once you have replied to those comments, I will approve your PR.🎉🎉

sunny u tagged shubham-bajaj but Thankyou so much for helping me @sahsisunny

shmbajaj commented on js/intro.js:
```diff
@@ -256,8 +260,7 @@ async function showSavedDetails() {
     if (usersRequest.status === 200) {
-      const usersDataList = await usersRequest.json();
-      const userData = usersDataList.data[0];
+      const userData = usersRequest.data.data[0];
```
`Suggestion:` You can use data like this.

```js
const userData = userRequest.data[0]
```

shubhamsigdar1 replied:
<img width="311" alt="image" src="https://user-images.githubusercontent.com/107163260/228872920-e9777631-7ef9-44f6-a735-db7b1846b0b5.png">

shubhamsigdar1 replied:
I am using make API call in two place
1. https://github.com/Real-Dev-Squad/website-www/pull/432/files#diff-b8034978785404c0d5db1ed2aa00f41fecf7baaf67f71b348a0a1806a5b24017R255

<img width="637" alt="image" src="https://user-images.githubusercontent.com/107163260/230154399-dd480b7b-48f3-427f-9eb7-d908d020e9e3.png">

2.  https://github.com/Real-Dev-Squad/website-www/pull/432/files#diff-b8034978785404c0d5db1ed2aa00f41fecf7baaf67f71b348a0a1806a5b24017R291

<img width="335" alt="image" src="https://user-images.githubusercontent.com/107163260/230155338-b0678de1-34e5-4b9e-87fa-9ae09ad4214c.png">


and you said to do this
check this @shmbajaj you write on the time of review...if i do this then i focus on particular api call..Is I am right??
 https://github.com/Real-Dev-Squad/website-www/pull/432#discussion_r1153365771

shubhamsigdar1 replied:
Talked to @shmbajaj and i am right...resolved

shmbajaj commented on js/intro.js:
```diff
@@ -317,7 +330,14 @@ async function showSavedDetails() {
+      ),
+    );
+    setTimeout(function () {
+      window.location.href = 'https://realdevsquad.com';
```
Use constant for url and timeInterval in seconds.

shmbajaj commented on js/intro.js:
```diff
@@ -317,7 +330,14 @@ async function showSavedDetails() {
+    window.addEventListener(
+      'click',
+      showToast(
+        '<i class="fa-solid fa-circle-exclamation"></i> something went wrong',
```
Instead of passing icon here, you can pass type and let the function **showToast** handle which icon to show.

shmbajaj commented on js/intro.js:
```diff
@@ -293,7 +281,7 @@ async function showSavedDetails() {
       loading.classList.add('hidden');
       setTimeout(() => {
-        alert('SuperUser You Write Wrong Url');
+        alert('SuperUser You Write Wrong userId');
```
Use constant.

shmbajaj commented on js/intro.js:
```diff
@@ -313,13 +301,17 @@ async function showSavedDetails() {
+      }
     } else {
-      if (userId == 'wrong url') {
+      if (!urlParams.has('id')) {
```
Aren't the conditions on line **307** and **314** are same?

RitikJaiswal75 replied:
+1

shubhamsigdar1 replied:
I was thinking that if normal user go to 'http://localhost:5000/intro.html`
he will get 404 page like superuser get

but if i removed this then normal user always get 401 page

shmbajaj commented on css/intro.css:
```diff
@@ -77,6 +77,12 @@
 }
 
 @media screen and (max-width: 500px) {
```
Why you have used **500px** as breakpoint here?

shubhamsigdar1 replied:
To make 404 small in mobile and in ui form when i go below 500 the text get cut

can u check this two video 
### intro-page UI
https://user-images.githubusercontent.com/107163260/228812084-79506418-0856-4d70-beb6-1119046a795f.mp4

### Add 404 page(url access by super user)
https://user-images.githubusercontent.com/107163260/228411773-e7d41b5d-4171-4514-854f-0f625befdfb1.mp4


To

shmbajaj commented on css/intro.css:
```diff
@@ -77,6 +77,12 @@
+  .user-input {
+    font: 300 17px 'Roboto';
+    width: 22rem;
+    line-height: 32px;
```
Either use `px` or `rem`.
Maintain Consistency.

shmbajaj commented on js/utils.js:
```diff
@@ -0,0 +1,32 @@
+  }
+}
+
+export function createElement({ type, classList = [], id }) {
```
What if id is empty or undefined?

shmbajaj replied:
You might need a utility function like this for `helpers/utils` functions, as they are globally used.

`Snippet`

```jsx
/**
 * @param value:
 * @returns boolean which returns
 * - `true` if value is empty or falsy
 * - `false`  if value is not empty or truthy
 */
function isEmpty(valueToCheck) {
  switch (typeof valueToCheck) {
    case "undefined":
      return true;
    case "string":
      return valueToCheck.trim().length === 0 || valueToCheck.length === 0;
    case "object":
      if (valueToCheck === null) {
        return true;
      } else if (Array.isArray(valueToCheck)) {
        return valueToCheck.length === 0;
      } else {
        return Object.keys(valueToCheck).length === 0;
      }
    case "number":
      return Number.isNaN(valueToCheck);
    default:
      return false;
  }
}
```

shmbajaj commented on js/utils.js:
```diff
@@ -0,0 +1,32 @@
+      ...options,
+    });
+    const res = {
+      data: await response.json(),
```
`Suggestion:`

```jsx
const {data} = await response.json();

const res = {
data,
status
}

```

shubhamsigdar1 commented on css/intro.css:
```diff
@@ -77,6 +77,12 @@
 
 @media screen and (max-width: 500px) {
+  .user-input {
+    font: 300 17px 'Roboto';
```
Thank you for pointing this...i changed only font-size

shmbajaj commented on js/utils.js:
```diff
@@ -14,19 +14,54 @@ export async function makeApiCall(
+ * - `false`  if value is not empty or truthy
+ */
+function isEmpty(valueToCheck) {
+  console.log(valueToCheck);
```
Remove console log.

shmbajaj reviewed (approved):
LGTM ✅

RitikJaiswal75 commented on intro.html:
```diff
@@ -0,0 +1,130 @@
+      </ul>
+    </nav>
+    <h1 class="loading">Loading...</h1>
+    <div class="not-authorized-page"></div>
+    <div class="not-found-page"></div>
+    <main class="intro-main"></main>
```
can this go in the same div, named content and we can add classes based on the js response? (Please take others' opinions on this as well)

shubhamsigdar1 replied:
@sahsisunny @shmbajaj what's your opinion on this??

sahsisunny replied:
Why are you making more complex, it is correct.

RitikJaiswal75 commented on intro.html:
```diff
@@ -0,0 +1,130 @@
+        >
+      </p>
+    </footer>
+    <script src="js/darkmode.js"></script>
```
Is this script being used here?

RitikJaiswal75 commented on js/intro.js:
```diff
@@ -0,0 +1,305 @@
+    classList: ['icon'],
+  });
+  img.src = 'img/intro-page/exclamation-circle.svg';
+  img.setAttribute('alt', 'exclamation-icon');
```
can we just keep it `exclamation-mark`

RitikJaiswal75 commented on js/intro.js:
```diff
@@ -0,0 +1,305 @@
+    classList: ['not-authorized-img'],
+  });
+  notAuthorizedImg.src = 'img/intro-page/page-not-authorized.png';
+  notAuthorizedImg.setAttribute('alt', 'not authorized page');
```
can we keep it as `not-authorized`

RitikJaiswal75 commented on js/intro.js:
```diff
@@ -0,0 +1,305 @@
+    type: 'p',
+    classList: ['input-label-dark'],
+  });
+  firstName.innerText = 'First Name';
```
can it be firstNameLabel (for easy understanding of code)

RitikJaiswal75 commented on js/intro.js:
```diff
@@ -0,0 +1,305 @@
+    classList: ['input-label-dark'],
+  });
+  firstName.innerText = 'First Name';
+  const renderFName = createElement({
```
this can become firstNameValue

RitikJaiswal75 replied:
Please check this on other variables as well

RitikJaiswal75 commented on js/intro.js:
```diff
@@ -0,0 +1,305 @@
+      generateNoDataFoundPage();
+      loading.classList.add('hidden');
+      setTimeout(() => {
+        alert('SuperUser You Write Wrong userId');
```
it can be a toast and message could be (Invalid user id entered)

RitikJaiswal75 commented on js/intro.js:
```diff
@@ -0,0 +1,305 @@
+
+(async function setAuth() {
+  try {
+    const res = await fetch(`${BASE_URL}/users/self`, {
+      method: 'GET',
+      credentials: 'include',
+      headers: {
+        'Content-type': 'application/json',
+      },
+    });
```
why aren't we using the makeAPICall function here?

RitikJaiswal75 commented on js/intro.js:
```diff
@@ -0,0 +1,305 @@
+  } catch (err) {
+    window.addEventListener('click', showToast('something went wrong', 6000));
+    setTimeout(function () {
+      window.location.href = HOME_URL;
```
why are we redirecting the user without informing?

shubhamsigdar1 replied:
i inform user using toast

ankushdharkar commented on js/utils.js:
```diff
@@ -0,0 +1,65 @@
+ * - `true` if value is empty or falsy
+ * - `false`  if value is not empty or truthy
+ */
+function isEmpty(valueToCheck) {
```
Do we really need such a verbose function?

shubhamsigdar1 replied:
https://github.com/Real-Dev-Squad/website-www/pull/432#discussion_r1153364091

ankushdharkar commented on js/utils.js:
```diff
@@ -0,0 +1,65 @@
+function createElement({ type, classList = [], id }) {
+  const element = document.createElement(type);
+  element.classList.add(...classList);
+  if (!id) {
+    element.id = isEmpty(id);
+  } else {
+    element.id = id;
+  }
```
Nit:
```suggestion
    element.id = id ?? isEmpty(id);
```

bhtibrewal commented on css/intro.css:
```diff
@@ -0,0 +1,98 @@
+
+  .not-found-text,
+  .not-authorized-text {
+    font-size: 1rem;
```
we shouldn't use rem for font-size I suggest you read on this

bhtibrewal replied:
read and ask which will be better, not just because the reviewer said so, for fonts we should use em

shubhamsigdar1 replied:
Thank you @bhtibrewal for pointing me to read about which unit we have to use for font-size
As I read `em` is relative to parent element and `rem` is relative to html element 
It good to use rem for font size 
when i read this two article- 
1. https://www.freecodecamp.org/news/why-use-rem-to-set-font-size-in-css/ 
2. https://medium.com/@dixita0607/css-units-for-font-size-px-em-rem-79f7e592bb97

bhtibrewal replied:
understanding the meaning and use of different units is important

shubhamsigdar1 replied:
okk @bhtibrewal thank you...I will try to careful from next time

vvaibhavdesai commented on css/intro.css:
```diff
@@ -0,0 +1,97 @@
+  animation: anim 5s linear forwards;
+}
+
+@keyframes anim {
```
rename this keyframe properly

vvaibhavdesai commented on intro.html:
```diff
@@ -0,0 +1,129 @@
+            My Profile
+          </a>
+        </li>
+        <hr class="line" />
```
where is the class line declared? 🤔

shubhamsigdar1 replied:
<img width="144" alt="image" src="https://user-images.githubusercontent.com/107163260/229697951-d7604a92-004a-44f6-9510-f0a283d4696c.png">

vvaibhavdesai commented on intro.html:
```diff
@@ -0,0 +1,129 @@
+      </ul>
+    </div>
+    <button id="scroll-to-top-btn">
+      <img src="./img/arrow_up.svg" alt="scroll to top button" />
```
your alt tag shall be the alt details for the image

shubhamsigdar1 replied:
Thank you for pointing out this button is not required

vvaibhavdesai commented on intro.html:
```diff
@@ -0,0 +1,129 @@
+      <img src="./img/arrow_up.svg" alt="scroll to top button" />
+    </button>
+    <nav>
+      <div class="hamburger">
```
from which file is this hamburger class coming ?

shubhamsigdar1 replied:
navbar.css file

vvaibhavdesai commented on js/utils.js:
```diff
@@ -0,0 +1,45 @@
+      status,
+    };
+    return res;
+  } catch (err) {
```
any reason why you are using `throw ` in catch block? as it will throw the error to its superior scope. are you wanting your superior scope function to do something very specific when this error is thrown?

shubhamsigdar1 replied:
I am using makeApiCall at two place but it is not using throw error very specific...So if it is not using should  I using console.log(err) here??

1. https://github.com/Real-Dev-Squad/website-www/pull/432/files#diff-b8034978785404c0d5db1ed2aa00f41fecf7baaf67f71b348a0a1806a5b24017R255

2.  https://github.com/Real-Dev-Squad/website-www/pull/432/files#diff-b8034978785404c0d5db1ed2aa00f41fecf7baaf67f71b348a0a1806a5b24017R291

vvaibhavdesai replied:
`console.error` should be a better alternative

shubhamsigdar1 replied:
vaibhav I am using `throw err` because utils.js file is a global file and it depend on particular user how they want to handle error in api call error...For ex: I am handle error by showing toast.

vvaibhavdesai replied:
but then are you even handling the error that is thrown in the superior scope?

shubhamsigdar1 replied:
yes I am handling that error if you see changes now...but also Have question correct me if i am wrong
Its depend on developer to developer that they want to show that error or not?

vvaibhavdesai commented on js/utils.js:
```diff
@@ -0,0 +1,45 @@
+ * - `true` if value is empty or falsy
+ */
+function isEmpty(valueToCheck) {
+  if (typeof valueToCheck === 'undefined') {
```
is this function complete?

shubhamsigdar1 replied:
> Make all the `png `into `svg`. Please format the code. It's difficult to read the logic inside the `generateSavedDetailsForm`. Attaching a SS below as an example of ![image](https://user-images.githubusercontent.com/64726586/229686678-cefd8b3c-efb7-4cc3-a5b6-8afc17b12ffa.png)

Hello vaibhav 
I am using two image as png in my particular file i.e. img/intro-page/page-not-authorized.png and page-not-found.png

and others image is globally used in all files including mine in navbar. So my question is should i convert png to svg only image of intro page ??

vvaibhavdesai replied:
make all your pngs into svgs

vvaibhavdesai replied:
also @shubhamsigdar1 what is this `isEmpty ` function going to check?  The code still looks the same to me

shubhamsigdar1 replied:
idEmpty function is checking Is `id` of element is undefine or null or not
if it is undefine then give `id=true` but i think i have to remove this function and do something like this...If I am wrong please correct me
```function createElement({ type, classList = [], id }) {
  const element = document.createElement(type);
  element.classList.add(...classList);
  element.id = id ?? true;
  return element;
}```

vvaibhavdesai reviewed (commented):
Make all the `png `into `svg`. 
Please format the code. It's difficult to read the logic inside the `generateSavedDetailsForm`. 
Attaching a SS below as an example of 
![image](https://user-images.githubusercontent.com/64726586/229686678-cefd8b3c-efb7-4cc3-a5b6-8afc17b12ffa.png)

vvaibhavdesai commented on js/intro.js:
```diff
@@ -0,0 +1,314 @@
+      }, 5000);
+    }
+  } catch (err) {
+    console.log(err);
```
use console.error here
